api/environments/environment_handler.go

```diff
@@ -170,68 +156,114 @@ func (eh EnvironmentHandler) getConfigurationStatusOfNamespace(namespace string)
-			}
+	if len(deploymentSummaries) == 1 {
+		environmentSummary.ActiveDeployment = deploymentSummaries[0]
+	}
 
-			environmentSummary := &environmentModels.EnvironmentSummary{
-				Name:   environment,
-				Status: environmentModels.Orphan.String(),
-			}
+	return environmentSummary, nil
+}
 
-			if len(deploymentSummaries) == 1 {
-				environmentSummary.ActiveDeployment = deploymentSummaries[0]
-			}
+func (eh EnvironmentHandler) getOrphanEnvironmentSummary(appName string, envName string) (*environmentModels.EnvironmentSummary, error) {
 
-			orphanedEnvironments = append(orphanedEnvironments, environmentSummary)
+	deploymentSummaries, err := eh.deployHandler.GetDeploymentsForApplicationEnvironment(appName, envName, latestDeployment)
+	if err != nil {
+		return nil, err
+	}
+
+	environmentSummary := &environmentModels.EnvironmentSummary{
+		Name:   envName,
+		Status: environmentModels.Orphan.String(),
+	}
+
+	if len(deploymentSummaries) == 1 {
+		environmentSummary.ActiveDeployment = deploymentSummaries[0]
+	}
+
+	return environmentSummary, nil
+}
+
+// getOrphanedEnvironments returns a slice of Summary data of orphaned Namespaces or RadixEnvironments
+func (eh EnvironmentHandler) getOrphanedEnvironments(appName string, radixApplication *v1.RadixApplication) ([]*environmentModels.EnvironmentSummary, error) {
+
+	orphanedEnvironments := make([]*environmentModels.EnvironmentSummary, 0)
+
+	for _, name := range eh.getOrphanedEnvNames(radixApplication) {
+		summary, err := eh.getOrphanEnvironmentSummary(appName, name)
+		if err != nil {
+			return nil, err
 		}
+
+		orphanedEnvironments = append(orphanedEnvironments, summary)
 	}
 
 	return orphanedEnvironments, nil
 }
 
+// getOrphanedEnvNames returns a slice of non-unique-names of orphaned Namespaces or RadixEnvironments
+func (eh EnvironmentHandler) getOrphanedEnvNames(app *v1.RadixApplication) []string {
+
+	envNames := make([]string, 0)
+	appLabel := fmt.Sprintf("%s=%s", kube.RadixAppLabel, app.Name)
+	namespacesInConfig := getNamespacesInConfig(app)
+	radixEnvironments, _ := eh.radixclient.RadixV1().RadixEnvironments().List(metav1.ListOptions{
+		LabelSelector: appLabel,
+	})
+
+	for _, re := range radixEnvironments.Items {
+		if isOrphaned(re.Name, namespacesInConfig) {
+			envNames = append(envNames, re.Spec.EnvName)
+		}
+	}
+
+	return envNames
+}
+
+func (eh EnvironmentHandler) namespaceExists(uniqueName string) (bool, error) {
+	_, err := eh.client.CoreV1().Namespaces().Get(uniqueName, metav1.GetOptions{})
+	return err == nil, err
+}
+
 func getNamespacesInConfig(radixApplication *v1.RadixApplication) map[string]bool {
 	namespacesInConfig := make(map[string]bool)
 	for _, environment := range radixApplication.Spec.Environments {
-		environmentNamespace := k8sObjectUtils.GetEnvironmentNamespace(radixApplication.Name, environment.Name)
-		namespacesInConfig[environmentNamespace] = true
+		uniqueName := k8sObjectUtils.GetEnvironmentNamespace(radixApplication.Name, environment.Name)
```

Comment:
I guess all references to namespace here is wrong. I.e. getNamespacesInConfig, namespacesInConfig etc.